Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Collections/GridObjectCollection.cs

```diff
@@ -454,16 +488,27 @@ public void OnValidate()
                 {
                     // Try to guess what the desired columns would be
                     int columnsGuess = Mathf.CeilToInt((float)nodeListCount / rows);
-                    string objectName = gameObject.name;
-                     if (gameObject.transform.parent != null) 
-                     {
-                         objectName += " (parent " + transform.parent.gameObject.name + ")";
-                     }
-                    Debug.Log("GridObjectCollection on " + objectName + " has layout ColumnsThenRows but columns are not specified. Most likely from asset upgrade to MRTK 2.2. Settings Columns property to " + nodeListCount + "/ " + rows + " = " + columnsGuess +". Check your asset to make sure GridObjectCollection has the correct values.");
-                    Columns = columnsGuess;
+                    string objectName = GetUserFriendlyName();
+                    Debug.Log("[GridObjectCollection upgrade to version 1] on " + objectName +
+                        ". Setting columns to " + nodeListCount + "/ " + rows + " = " + columnsGuess
+                        + ". Check asset to make sure GridObjectCollection has the correct values.");
+                    columns = columnsGuess;
                 }
             }
         }
 
+        private string GetUserFriendlyName()
+        {
+            string objectName = gameObject.name;
+            if (gameObject.transform.parent != null)
+            {
+                objectName += " (parent " + transform.parent.gameObject.name + ")";
```

Comment:
Nit: For some of these, consider using string interpolation which can make for a little neater looking stuff:

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated

Also mentioning this because it's more common in some other newer languages.

Reply:
(Especially with the Debug.Log of the one on line 491, the reduction in ""'s could possibly increase the readability of the final format